core/google/cloud/client.py

```diff
@@ -135,7 +135,7 @@ def __init__(self, credentials=None, _http=None):
                     credentials, google.auth.credentials.Credentials)):
             raise ValueError(_GOOGLE_AUTH_CREDENTIALS_HELP)
         if credentials is None and _http is None:
-            credentials = get_credentials()
+            credentials, _ = google.auth.default()
```

Comment:
hrm, this throws away the project. Does this match the behavior we're seeing today?

I would very much like to unify all of this - it seems we could probably fold all of the classes here into one and have a flag on whether a `project` is required or not. But you don't have to do this here, just thinking out loud.